app/org/maproulette/controllers/api/TaskController.scala

```diff
@@ -616,22 +616,39 @@ class TaskController @Inject() (
     * @param reviewStatus The review status id to set the task's review status to
     * @param comment An optional comment to add to the task
     * @param tags Optional tags to add to the task
+    * @param newTaskStatus Optional new taskStatus to change the task's status
     * @return 400 BadRequest if task with supplied id not found.
     *         If successful then 200 NoContent
     */
   def setTaskReviewStatus(
       id: Long,
       reviewStatus: Int,
       comment: String = "",
-      tags: String = ""
+      tags: String = "",
+      newTaskStatus: String = ""
   ): Action[AnyContent] = Action.async { implicit request =>
     this.sessionManager.authenticatedRequest { implicit user =>
-      val task = this.dal.retrieveById(id) match {
+      var task = this.dal.retrieveById(id) match {
         case Some(t) => t
         case None =>
           throw new NotFoundException(s"Task with $id not found, cannot set review status.")
       }
 
+      // If the mapper wants to change the task status while revising the task after review
```

Comment:
Same comment as above for this section. Also this code in general looks very similar, so may want to see if there is any way to combine it.

Reply:
Refactored to remove var